R/calcGAINSEmi.R

```diff
@@ -64,27 +63,30 @@ calcGAINSEmi <- function(subtype = "emissions") {
     ef_gains[, 2005, ]  <- setYears(ef_gains[, 2010, ])
     emi_gains[, 2005, ] <- setYears(emi_gains[, 2010, ])
 
-    w <- emi_gains
     x <- ef_gains
+
     unit <- "Mt/TWa"
     description <- "Emission factors from GAINS for all CEDS setors execpt aviation and shipping"
-    weight <- w
-  }
 
-  if (subtype == "emissions_starting_values") {
+    weight <- emi_gains
+
+    # do not throw a warning for zero weights, as they seem to be intended
+    aggregationArguments <- list(zeroWeight = "allow")
```

Comment:
Allow zero weight to suppress warning and not change input data (author Sebastian Rauner is no longer with us)